backend/sportsmatch/src/main/java/com/sportsmatch/auth/JwtAuthFilter.java

```diff
@@ -39,7 +40,14 @@ protected void doFilterInternal(
     }
 
     final String jwt = authHeader.substring(7);
-    final String userEmail = jwtService.extractUserName(jwt);
+
+    final String userEmail;
+    try {
+      userEmail = jwtService.extractUserName(jwt);
+    } catch (Exception e) {
+      response.setStatus(HttpStatus.UNAUTHORIZED.value());
```

Comment:
Does this let you see public endpoints when you have invalid token?

I think it may be better to simply not add anything to the user context (instead of returning 401) and thus still allowing access to public endpoints.